File: js/푸드파이트 대회.js
function solution(food) {
    var arr = [];
    for (let i=1; i<food.length; i++) {
        let cnt = Math.floor(food[i] / 2);
        for (let j=0; j<cnt; j++) {
            arr.push(i);
        }
    }
    // 주의사항: 그냥 arr.reverse()를 하게되면 arr자체가 뒤집어짐 -> 원본배열이 수정됨
    // reverse()의 영향을 받지 않기 위해 arr 배열을 복사한 뒤 사용해야 함!!(전개연산자 사용)
    var converted_arr = [...arr].reverse();
    return arr.join('') + '0' + converted_arr.join('');
}

File: python/푸드파이트 대회.py
def solution(food):
    arr = []
    for i in range(1,len(food)):
        cnt = food[i] // 2
        arr.append(str(i) * cnt)
    converted_arr = sorted(arr, reverse=True)

    return ''.join(arr) + '0' + ''.join(converted_arr)